Source/inv.cpp

```diff
@@ -581,8 +581,10 @@ void CheckInvPaste(int pnum, Point cursorPosition)
 	}
 	CalcPlrInv(player, true);
 	if (pnum == MyPlayerId) {
-		if (cn == CURSOR_HAND && !IsHardwareCursor())
+		if (cn == CURSOR_HAND && !IsHardwareCursor()) {
+			Size cursSize = GetInvItemSize(MyPlayer->HoldItem._iCurs + CURSOR_FIRSTITEM);
 			SetCursorPos(MousePosition + Displacement(cursSize / 2));

```

Comment:
This relies on HoldItem staying valid after being emptied. Using `itemSize * INV_SLOT_HALF_SIZE_PX` here has the same effect (or create a variable like hotPixelOffset in stash.cpp).